src/main/java/bio/overture/ego/repository/queryspecification/UserPermissionSpecification.java

```diff
@@ -16,32 +16,7 @@
 
 package bio.overture.ego.repository.queryspecification;
 
-import static bio.overture.ego.model.enums.JavaFields.ID;
-import static bio.overture.ego.model.enums.JavaFields.OWNER;
-import static bio.overture.ego.model.enums.JavaFields.POLICY;
+import lombok.extern.slf4j.Slf4j;
 
-import bio.overture.ego.model.entity.Policy;
-import bio.overture.ego.model.entity.UserPermission;
-import java.util.UUID;
-import javax.persistence.criteria.Join;
-import lombok.NonNull;
-import org.springframework.data.jpa.domain.Specification;
-
-public class UserPermissionSpecification extends SpecificationBase<UserPermission> {
-
-  public static Specification<UserPermission> withPolicy(@NonNull UUID policyId) {
-    return (root, query, builder) -> {
-      query.distinct(true);
-      Join<UserPermission, Policy> applicationJoin = root.join(POLICY);
-      return builder.equal(applicationJoin.<Integer>get(ID), policyId);
-    };
-  }
-
-  public static Specification<UserPermission> withUser(@NonNull UUID userId) {
-    return (root, query, builder) -> {
-      query.distinct(true);
-      Join<UserPermission, Policy> applicationJoin = root.join(OWNER);
-      return builder.equal(applicationJoin.<Integer>get(ID), userId);
-    };
-  }
-}
+@Slf4j
+public class UserPermissionSpecification extends AbstractPermissionSpecification {}
```

Comment:
bring back 
https://github.com/overture-stack/ego/blob/5a5ce1e198/src/main/java/bio/overture/ego/repository/queryspecification/UserPermissionSpecification.java

the owner field is of type User for UserPermissionSpecification and of type Group for GroupPermissionSpecification, so the join that is done in AbstractPermissionSpecification must define the owner type explicitly.